applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c

```diff
@@ -1603,135 +1312,228 @@ int unicast_client_config_get(struct bt_conn *conn, enum bt_audio_dir dir, uint3
 void unicast_client_conn_disconnected(struct bt_conn *conn)
 {
 	int ret;
-	struct stream_index idx;
 
-	ret = device_index_get(conn, &idx);
+	ret = srv_store_lock(CAP_PROCED_SEM_WAIT_TIME_MS);
+	if (ret < 0) {
+		LOG_ERR("%s: Failed to lock server store: %d", __func__, ret);
+		return;
+	}
+
+	struct server_store *server = NULL;
+
+	ret = srv_store_from_conn_get(conn, &server);
 	if (ret) {
-		LOG_WRN("Unknown connection disconnected");
-	} else {
-		disconnected_cleanup(idx);
+		LOG_DBG("%s: Unknown server", __func__);
+		srv_store_unlock();
+		return;
 	}
+
+	/* Clear endpoint information */
+	memset(&server->snk.eps, 0, sizeof(server->snk.eps));
```

Comment:
Discuss. This may be done twice now. 
Why can we not use the built-in function to clear the server?

Reply:
Will have a look

Reply:
Switched to `clear_by_conn`